enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,226 @@
+---
+title: replace-prometheus-adapter-with-metrics-server
+authors:
+  - "@slashpai"
+reviewers:
+  - "@simonpasquier"
+  - "@dgrisonnet"
+  - "@openshift/openshift-team-monitoring"
+approvers: 
+  - "@simonpasquier"
+  - "@openshift/openshift-team-monitoring"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-06-20
+last-updated: 2023-06-26
+tracking-link:
+  - https://issues.redhat.com/browse/MON-3153
+---
+
+# Replace prometheus-adapter with metrics-server
+
+## Summary
+
+This proposal suggests replacing the [Prometheus Adapter](https://github.com/kubernetes-sigs/prometheus-adapter) currently used for implementing the resource [metrics API](https://github.com/kubernetes/metrics) in OpenShift with the [Metrics Server](https://github.com/kubernetes-sigs/metrics-server).
+
+## Motivation
+
+For vertical and horizontal pod autoscaling, a resource metrics API implementation is necessary. Currently, OpenShift relies on the Prometheus stack and Prometheus adapter managed by the Cluster Monitoring Operator (CMO).
+
+Reasons to consider to moving to Metrics Server are:
+
+1. **Upcoming Archival of prometheus-adapter:** The Prometheus Adapter project will soon be archived due to difficulties in maintaining the project, making it less desirable for long-term usage.
+
+1. **Simplified Monitoring Architecture:** By replacing Prometheus Adapter with Metrics Server, we can reduce the complexity and maintenance overhead of the monitoring system in OpenShift since `metrics-server` does not rely on Prometheus.
+
+1. **Lightweight Solution:** Deploying and configuring Prometheus Adapter requires additional effort compared to Metrics Server. Prometheus Adapter is tightly coupled to Prometheus and this creates a bottleneck for allowing monitoring as  optional feature. See [MON-3152](https://issues.redhat.com/browse/MON-3152).
+
+1. **Better Accuracy:**  By default, the Metrics Server takes into account the timestamps of metric samples, allowing for accurate tracking of resource utilization over time. This means you can rely on the Metrics Server to provide a more precise understanding of how resources are being utilized by your applications.
+
+### User Stories
+
+#### Story 1
+
+* Inorder to provide [optional built-in monitoring](https://issues.redhat.com/browse/MON-3152) feature we need to provide a way for customers to do HPA other than through Prometheus based Prometheus Adapter.
+
+#### Story 2
+
+* As a cluster admin, I want a more lightweight and efficient solution for monitoring resource metrics in my cluster, so that I can optimize cluster performance and resource allocation.
```

Comment:
I'm not sure about the persona. The cluster admin probably doesn't care so much about which of metrics-server or prometheus-adapter is used. But they should care about the improved accuracy.

Reply:
+1 The accuracy improvements are a nice positive. And this will allow us to drop our workaround (dedicated resource metrics service monitor) and reduce config complexity.
The resource improvements will only really be realized when we have option in-cluster monitoring.